armi/reactor/blocks.py

```diff
@@ -1212,15 +1215,17 @@ def mergeWithBlock(self, otherBlock, fraction):
         into a full block.
         """
 
+        numDensities = self.getNumberDensities()
```

Comment:
Going even further, this method and the `self.getNuclides()` call both have to run `getNuclides()` which is redundant. I recommend:

```
nuclideNames = self.getNuclides();
numDensities = self.getNuclideNumberDensities()

for nucName, ndens in zip(nucNames, nucDensities):
   set.setNum...
```

Oohh and if you want to get crazy you can use `setNumberDensities` to speed things up on the setting side. Though really that could be a different diff.